plugin/builtin/helmchartinflationgenerator/HelmChartInflationGenerator.go

```diff
@@ -258,33 +287,68 @@ func (p *plugin) Generate() (rm resmap.ResMap, err error) {
 	// try to remove the contents before first "---" because
 	// helm may produce messages to stdout before it
 	stdoutStr := string(stdout)
-	if idx := strings.Index(stdoutStr, "\n---\n"); idx != -1 {
+	if idx := strings.Index(stdoutStr, "---"); idx != -1 {
```

Comment:
Won't this potentially delete the first document if a header isn't used?

FWIW this is the doc splitting code the byte reader uses:

https://github.com/kubernetes-sigs/kustomize/blob/14947e449bde0785cd6cc3ee90562095b34f1985/kyaml/kio/byteio_reader.go#L170-L196

As you can see, it uses a regex that is attempting to accommodate comments.

Reply:
You might be right, in any case I think using the regex in the doc splitter code that you linked makes sense. In my most recent commit, I made the `splitDocuments` function exportable and used it directly. Now it looks like this:

```
	documents, err := kio.SplitDocuments(string(stdout))
	if err != nil {
		return nil, err
	}
	var nodes []*kyaml.RNode
	for _, doc := range documents {
		n, err := kyaml.Parse(doc)
		if err != nil {
			continue
		}
		nodes = append(nodes, n)
	}

	if len(nodes) != 0 {
		return p.h.ResmapFactory().NewResMapFromRNodeSlice(nodes)
	}
```

This way, we won't be skipping any documents unless we get any error while parsing it. Thanks for the link!